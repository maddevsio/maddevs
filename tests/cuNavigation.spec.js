import { mount } from '@vue/test-utils'
import CuNavigation from '../client/components/Blog/CuNavigation'

describe('cuNavigation', () => {
  let wrapper

  const clusterPosts = [
    {
      chapter_name: [
        {
          spans: [],
          text: 'Chapter 1',
          type: 'heading1',
        },
      ],
      cu_post: {
        id: 'YAGi7REAACMAgV8d',
        isBroken: false,
        lang: 'en-us',
        link_type: 'Document',
        slug: 'adadadad',
        tags: [],
        type: 'customer_university',
        uid: 'cu-test',
      },
    },
    {
      chapter_name: [
        {
          spans: [],
          text: 'Chapter 2',
          type: 'heading1',
        },
      ],
      cu_post: {
        id: 'YAGi7REAACMAgV8d',
        isBroken: false,
        lang: 'en-us',
        link_type: 'Document',
        slug: 'adadadadaa',
        tags: [],
        type: 'customer_university',
        uid: 'cu-test-2',
      },
    },
  ]

  beforeEach(() => {
    wrapper = mount(CuNavigation, {
      stubs: ['router-link'],
      propsData: {
        clusterPosts,
        cluster: {
          items: clusterPosts,
          primary: {
            cluster_name: 'Pricing strategies in custom software development',
            read_more_text: 'Read more about pricing strategies',
          },
        },
        id: '123qweasd',
      },
      mocks: {
        $prismic: {
          asText: text => text[0].text,
        },
      },
    })
  })

  it('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
