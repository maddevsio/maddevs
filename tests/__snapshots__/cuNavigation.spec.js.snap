// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cuNavigation renders correctly 1`] = `
<div
  class="cluster-navigation"
>
  <div
    class="container"
  >
    <div
      class="cluster-navigation__title"
    >
      
    </div>
     
    <div
      class="cluster-navigation__inner"
    >
      <ol
        class="cluster-navigation__column cluster-navigation__column--left"
      >
        <router-link-stub
          class="cluster-navigation__nav-item"
          to="/customer-university/cu-test/"
        >
          <div
            class="index"
          >
            1.
          </div>
           
          <div
            class="label"
          >
            Chapter 1
          </div>
        </router-link-stub>
      </ol>
       
      <ol
        class="cluster-navigation__column cluster-navigation__column--right"
        start="2"
      >
        <router-link-stub
          class="cluster-navigation__nav-item"
          to="/customer-university/cu-test-2/"
        >
          <span
            class="index"
          >
            2.
          </span>
           
          <div
            class="label"
          >
            Chapter 2
          </div>
        </router-link-stub>
      </ol>
    </div>
  </div>
</div>
`;
